src/cpp/server/secure_server_credentials.cc

```diff
@@ -103,6 +104,13 @@ void AuthMetadataProcessorAyncWrapper::InvokeProcessor(
   }
   auto consumed_md_data = consumed_md.empty() ? nullptr : &consumed_md[0];
   auto response_md_data = response_md.empty() ? nullptr : &response_md[0];
+
+  // Instantiate GrpcLibraryCodegen to call grpc_init()/grpc_shutdown().
+  // We need this here because this whole method will be invoked asynchronously,
+  // potentially after all grpc_shutdown()s have been called. Because "cb" below
+  // is an arbitrary callback, we need to ensure it's called with the grpc
+  // library properly initialized.
+  GrpcLibraryCodegen init;
```

Comment:
That won't help. I need to improve the comment, but what happens is that a callback can be scheduled, the instance of AuthMetadataProcessorAyncWrapper be destroyed, the callback run and kaboom. Note that there's no "join" method or anything similar in the thread pools used. The moment a callback is scheduled it's effectively a detached thread.

In the case of the end2end test, the callback in question belongs to a runaway thread from a previous test case (it can be seen with enough logging), but it could still happen in real code based on how detached the lifetime of scheduled threads is from that of the objects doing the scheduling.

A more effective way of handling this would be with reference counting, making the scheduled cb responsible for unreffing.